lib/rubocop/cop/rspec/capybara/specific_matcher.rb

```diff
@@ -0,0 +1,73 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module RSpec
+      module Capybara
+        # Checks for if there is a more specific matcher offered by Capybara.
+        #
+        # @example
+        #
+        #   # bad
+        # expect(page).to have_css('button')
+        # expect(page).to have_css('button', exact_text: 'foo')
+        # expect(page).to have_css('button[disabled]', exact_text: 'foo')
+        # expect(page).to have_css('a')
+        # expect(page).to have_css('a.cls', exact_text: 'foo')
+        # expect(page).to have_css('table')
+        # expect(page).to have_css('table.cls')
+        #
+        #   # good
+        # expect(page).to have_button
+        # expect(page).to have_button('foo')
+        # expect(page).to have_button('foo', disabled: true)
+        # expect(page).to have_link
+        # expect(page).to have_link('foo', class: 'cls')
+        # expect(page).to have_table
+        # expect(page).to have_table(class: 'cls')
+        #
+        class SpecificMatcher < Base
+          RESTRICT_ON_SEND = %i[have_css have_no_css has_css has_no_css].freeze
+          SPECIFIC_MATCHER = {
+            button: 'button',
+            a: 'link',
+            table: 'table'
+          }.freeze
+          MSG_HAVE_CSS = 'Use `have_%<matcher>s` instead of `have_css`.'
+          MSG_HAVE_NO_CSS = 'Use `have_no_%<matcher>s` ' \
+                              'instead of `have_no_css`.'
+          MSG_HAS_CSS = 'Use `has_%<matcher>s` instead of `has_css`.'
+          MSG_HAS_NO_CSS = 'Use `has_no_%<matcher>s` instead of `has_no_css`.'
```

Comment:
What are the cases when the use of `has_...` matchers forms a correct English sentence?

Reply:
Oh, `has_...` method ends with `? ` at the end.
https://www.rubydoc.info/gems/capybara/Capybara/Node/Matchers#has_css%3F-instance_method

I've improved the definition of this message because each time the number of supported methods increases, the definition of this message also increases.